pkg/webhooks/generate/generate.go

```diff
@@ -111,20 +111,41 @@ func retryApplyResource(client *kyvernoclient.Clientset,
 		gr := kyverno.GenerateRequest{
 			Spec: grSpec,
 		}
-		gr.SetGenerateName("gr-")
+
 		gr.SetNamespace(config.KubePolicyNamespace)
 		// Initial state "Pending"
 		// TODO: status is not updated
 		// gr.Status.State = kyverno.Pending
 		// generate requests created in kyverno namespace
-		if action == v1beta1.Create {
-			_, err = client.KyvernoV1().GenerateRequests(config.KubePolicyNamespace).Create(&gr)
-		}
-		if action == v1beta1.Update {
-			gr.SetLabels(map[string]string{
-				"resources-update": "true",
-			})
-			_, err = client.KyvernoV1().GenerateRequests(config.KubePolicyNamespace).Update(&gr)
+		isExist := false
+		if action == v1beta1.Create || action == v1beta1.Update {
+			grList, err := client.KyvernoV1().GenerateRequests(config.KubePolicyNamespace).List(metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			for _, v := range grList.Items {
+				if grSpec.Policy == v.Spec.Policy && grSpec.Resource.Name == v.Spec.Resource.Name && grSpec.Resource.Kind == v.Spec.Resource.Kind && grSpec.Resource.Namespace == v.Spec.Resource.Namespace {
+
+					gr.SetLabels(map[string]string{
+						"resources-update": "true",
+					})
+					v.Spec.Context = gr.Spec.Context
+					v.Spec.Policy = gr.Spec.Policy
+					v.Spec.Resource = gr.Spec.Resource
+					_, err = client.KyvernoV1().GenerateRequests(config.KubePolicyNamespace).Update(&v)
+					if err != nil {
```

Comment:
@evalsocket Can you fix this issue?
>[golangci] reported by reviewdog 🐶
G601: Implicit memory aliasing in for loop. (gosec)